src/environment.js

```diff
@@ -122,9 +123,11 @@ if (process.env.NODE_ENV !== 'production') {
 
 export const defaultEthNode =
   getDefaultEthNode() || networkConfig.nodes.defaultEth
+const defaultEthProvider = new Web3.providers.WebsocketProvider(defaultEthNode)
+
 export const web3Providers = {
-  default: new Web3.providers.WebsocketProvider(defaultEthNode),
-  wallet: window.web3 && window.web3.currentProvider,
+  default: defaultEthProvider,
+  wallet: provider(['injected', 'frame', defaultEthNode]),
```

Comment:
Having the provider as a default wallet could be useful in some cases (non-injected local node), but it would also make the [wallet check](https://github.com/aragon/aragon/blob/f896c00d40c02542ffe1418064dc868022ebbc36/src/App.js#L58) pass every time, even when using a read-only provider.

Maybe we should not have it by default and have a setting for that, below the provider URL, for users that know what they are doing (“[✓] Sign transactions with this provider”).

Reply:
I think the provider gives us some utility methods to check if it's connected, so we could modify the wallet check to use that instead.

I'd hesitate to add the `defaultEthNode` as a fallback, since that's unlikely to be unlocked (unless you control that node).

Reply:
Okay, since we will also use this to check connectivity, can someone more familiarized with this take it from here? Shouldn't take a lot of time, but will be easier if someone familiar with it does it @sohkai @bpierre

Reply:
Yes I can take it from here @luisivan, I’ll have a look later today 👍

Reply:
Thanks a lot! :+1: